src/util/helpers/helpers.cpp

```diff
@@ -417,18 +417,19 @@ std::string GenerateRandomString(size_t length)
 	return GenerateRandomString(length, kCharacters);
 }
 
-std::string GenerateRandomString(size_t length, std::string_view characters)
+std::string GenerateRandomString(const size_t length, const std::string_view characters)
 {
 	assert(!characters.empty());
-	std::stringstream result;
+	std::string result;
+	result.reserve(length);
 
 	std::random_device rd;
 	std::mt19937 gen(rd());
 	std::uniform_int_distribution<decltype(characters.size())> index_dist(0, characters.size() - 1);
 	for (uint32_t i = 0; i < length; ++i)
 	{
-		result << characters[index_dist(gen)];
+		result += characters[index_dist(gen)];
```

Comment:
And replace this with `result[i] = characters[index_dist(gen)];`
That lets you avoid the overhead that comes with appending (size tracking and checks)

Reply:
I've changed this to std::generate_n(), so that no loop is needed at all.